### PR TITLE
Enhance Job Status Results

### PR DESCRIPTION
**Pull request by joe-bell**
## Description

Adds a new Job "status" UI; with more detail in the Job View, and an updated non-detailed status within table view 

As part of this task, I also added a new `Grid` component to our Design System, and an orchest-specific `DescriptionList` component to take care of common "term/details" layout (in a more semantic manner)

### Checklist

- [x] The documentation reflects the changes.
- [x] I have manually tested the application to make sure the changes don’t cause any downstream issues, which includes making sure `./orchest status --ext` is not reporting failures when Orchest is running.

- [x] In case I changed one of the services’ `models.py` I have performed the appropriate database migrations.

### CODE REVIEW

**Comment by yannickperrenet** [comment]
First off, awesome work!

I happened to notice that the vertical alignment is off in the table view of Jobs (the "Started..." text is slightly above center):

image

File: services/orchest-webserver/client/src/views/JobView.jsx [hunk 1]
```diff
@@ -1,120 +1,257 @@
...
+                  value: count.SUCCESS,
+                },
+              ]}
+            />
+          </Box>
+        )
+      }
+      title={
+        {
+          DRAFT: "This job is still being drafted",
```
Comment (yannickperrenet) [inline services/orchest-webserver/client/src/views/JobView.jsx:137]: When creating a cron-scheduled job, then this `DRAFT` message is shown. Even though the job is actually already running.

**Comment by ricklamers** [comment]
This is an awesome enhancement of the job status indicator.

Minor nitpick: I think the sizing in this view could be a bit more balanced.

image

I think the difference between the top left icon and the icons bottom right is a bit too big.

**Comment by ricklamers** [comment]
Could you merge in master so we can test with the working directory fix in there?

**Comment by ricklamers** [comment]
Related to my other comment about size balance, I feel like the typography sizing is a bit off here as well:

image

Would it be an idea to have smaller text sizes for the `no parameter hint` in the list view?

File: services/orchest-webserver/client/src/components/Status.jsx [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        DRAFT: (
+          <React.Fragment>
+            <IconDraftOutline css={{ color: "$gray500" }} />
+            Draft
+          </React.Fragment>
+        ),
+        STARTED: (
+          <React.Fragment>
+            <IconClockOutline css={{ color: "$warning" }} />
+            Started…
```
[the remark this answers is not in the document]
Comment (ricklamers) [reply services/orchest-webserver/client/src/components/Status.jsx:52]: I like the proposal. It’s hard to find the right words, but what you’ve proposed sounds the most logical to me.

**Comment by joe-bell** [comment]
> First off, awesome work!
> 
> I happened to notice that the vertical alignment is off in the table view of Jobs (the "Started..." text is slightly above center):
> 
> image

@yannickperrenet Good spot, I'll get onto this!

---

> This is an awesome enhancement of the job status indicator!
> 
> Minor nitpick: I think the sizing in this view could be a bit more balanced.
> 
> image
> 
> I think the difference between the top left icon and the icons bottom right is a bit too big.
> 
> One idea would be to add the icon to the Status label instead. Or is that silly?

@ricklamers I think making the icon size a bit smaller makes more sense! It's worth notice that when two lines are visible in the status it's not so drastic, but I agree it could be smaller

---

> Could you merge in master so we can test with the working directory fix in there?

Sure thing

---

> Related to my other comment about size balance, I feel like the typography sizing is a bit off here as well:
> 
> image
> 
> Would it be an idea to have smaller text sizes for the `no parameter hint` in the list view?

@ricklamers I think this is probably out of scope for this ticket considering it was added previously, but I agree – things feel a bit "off". My suggestion would be that we'll follow this up in additional issue/PR to address whether or not a smaller `Alert` variant is best or other typographic changes

File: services/orchest-webserver/client/src/components/Status.jsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (joe-bell) [reply services/orchest-webserver/client/src/components/Status.jsx:52]: So just to clarify, this component is a "dumb" component - the actual logic for this is applied later on in the place it's used

We have a `PENDING` status already (below on line 55 of this file), so perhaps it's just a matter of changing our existing JobView to use this `status` instead

File: services/orchest-webserver/client/src/views/JobView.jsx (hunk 1 above)
Comment (yannickperrenet) [reply services/orchest-webserver/client/src/views/JobView.jsx:137]: #### To reproduce

1. Create a new job
2. Make sure it is a "cron job"
3. Press "Run job"
4. Select the job in the jobs list

Now you can see at the top that the "Status" is: "This job is still being drafted"

Comment (yannickperrenet) [reply services/orchest-webserver/client/src/views/JobView.jsx:137]: #### To reproduce

1. Create a new job
2. Make sure it is a "cron job"
3. Press "Run job"
4. Select the job in the jobs list

Now you can see at the top that the "Status" is: "This job is still being drafted"

File: services/orchest-webserver/client/src/components/Status.jsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (yannickperrenet) [reply services/orchest-webserver/client/src/components/Status.jsx:52]: > this component is a "dumb" component

The fact that I placed the comment on this specific line had nothing to do with the component, it was meant as a note on "Started..." itself.

> so perhaps it's just a matter of changing our existing JobView to use this status instead

Could very well be, I just thought it was most reasonable to split the current "Starting..." into two different statuses for the cron jobs. I am sure you can come up with reasonable wording and implementation there ;)

[the remark this answers is not in the document]
Comment (joe-bell) [reply services/orchest-webserver/client/src/components/Status.jsx:52]: I think my suggestion would be "Running…" with a green clock icon for cron jobs – not sure what anyone else thinks?

**Comment by yannickperrenet** [comment]
I do really like that donuts icon. Can't we use that for the cron jobs as well? I think it should be obvious from the job list (knowing that something is a cron job) that one of its runs has failed.

**Comment by joe-bell** [comment]
I definitely agree that cron jobs should be clearer at indicating that a job run has failed, but I don't think this donut would look too good at a smaller size (plus job runs can't be dismissed, so it might be confusing if a user has already seen the error but the donut indicates there's an issue)

My suggestion would be to stick with what we have for now, but perhaps explore the UX of cron jobs further at another date – as there's probably a lot of ideas we could explore here

File: services/orchest-webserver/client/src/views/JobView.jsx (hunk 1 above)
Comment (joe-bell) [reply services/orchest-webserver/client/src/views/JobView.jsx]: Got it, should be fixed

File: services/orchest-webserver/client/src/components/Status.jsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (joe-bell) [reply services/orchest-webserver/client/src/components/Status.jsx]: Resolved ✅

**Comment by joe-bell** [comment]
@ricklamers / @yannickperrenet I've addressed all changes (excluding the `no parameter hint` typography which I'll raise a separate ticket to explore)

**Comment by yannickperrenet** [comment]
One last thing I noticed.

This is applicable only to scheduled jobs and cron jobs.
* When starting a scheduled job it rightfully states "Pending..." until it is running "Running..."
* For a cron job, I think it should be in "Pending..." always unless a pipeline run is going on in which case it should be "Running...". This mimics the behavior of the scheduled jobs

**Comment by joe-bell** [comment]
@yannickperrenet I'm all on board with this **but** I think the right thing to do here would be to change the backend response for cron jobs to provide this logic (it should only return `STARTED` if it's running, and `PENDING` if not)

The front-end is only communicating what the backend is telling it, so by changing that state logic on the front-end we lose that single source of truth

cc @fruttasecca

**Review by yannickperrenet — APPROVED** [review]

**Pull request merged**